### ISSUE TITLE
Recipe Issues

### ISSUE THREAD

**Issue by Grimsrude** [MSandro/sky_fabrication3#68]
3 recipe issues I noticed starting from scratch on [0.5.7a]

1.) Alloy Forge creates the Brass Ingot from Create which is incompatible with anything - it should be making the Brass Ingot from TheTechReborn. 
2.) Similarly with Nickel - the ingot created is from the wrong mod. Can't be used. 
3.) The Rolling Mill from Create has 2 recipes for Copper Plate, which makes it impossible to make Copper Wire. Need to change either Copper Curved Plates recipe or Copper Wires

Edit: 
4.) Alloy Forge also can't make Tin Ingots for the same reasons. Thankfully the Bronze recipe is correct.

**Comment by KonSola5** [comment]
For the point 3: you are supposed to use a Steel Wiremill to craft MI Copper Wires in order to craft Copper Spools.

Here are more recipe issues, which are more leaned towards endgame:
(Edit: 🔧 means this will be fixed in 0.6.0)

1, 2, 3, 4 - 🔧

5.) 🔧 Record Fragments are only craftable using "C418 - 13" Disc - this has been an issue since Sky FABRICation 2 (!)
Shouldn't it just use the "minecraft:music_discs" tag?

6.) Infinity Catalysts require two unautomatable items: Ultimate Fish and Chips and Glitch Ingots.

The first one is a cool concept, however no mods offer automatable way to obtain GoFish fish.
Industrial Revolution's Fisher only fishes from loot tables: "gameplay/fishing/fish", "gameplay/fishing/junk" and "gameplay/fishing/treasure" - the list of loot tables is hardcoded into this mod's block.
So the only way is sitting for couple dozen hours, manually fishing with the Lure VI Bedrock Fishing Rod. Absolutely not fun.

The other one is obtained by defeating the System Glitch over, and over, and over,... Also, maually or AFK if you are creative with using the Create mod. One Glitch drops three Physicaly-Condensed Glitch Matters, 1 Glitch Ingot needs 8 of them, so the Infinity Ingot requires killing a Self-Aware System Glitch 30 times, and so the Quantum Armor + Sword requires killing the System Glitch 2259 times. Again, not fun at all.

7.) The Infinity Booster Cards' recipe is a bit misleading - looking at the recipe, it doesn't contain any endgame items.
Then you are looking at the Massive Table's recipe - oh, it requires a Neutronium Nugget.

8.) 🔧 Needlessly "gated" Petal Apothecary - requiring rare mob drops (Cursed Seeds) to craft a progression item is not a good recipe design.
This is the case only with a cobblestone Petal Apothecary - other ones do not require Cursed Seeds.

9.) 🔧 Needlessly complicated Netherite Mesh - it is only used for one purpose - getting Uraninite. At this point, you are probably running a Electric Quarry on full blast, and at a certain point MI's EU generation overpowers any E generator.
And so, it's useless.

10.) Large Compact Machine requires an unobtainable in Survival - Polished Spectral Shard Block. (btw, this is the "Spectrum Endgame" item, which is in the inaccessible, unfinished part of the mod)

11.) Giant Compact Machine requires an unautomatable, non-renewable item - Block of Fiery Metal - obtaining of which requires you to kill Hydras and Ur-Ghasts.

12.) Maximum Compact Machine requires an UU-Matter Bucket. At this point it's completely useless since you have automated everything required for the endgame, and compacting is not required.

(Compact Machines are bugged anyway, so they are completely unviable, even if their crafting cost is reduced)

13.) A leftover recipe from Sky FABRICation 2 - for Advanced Circuits. In SF2, it gated you from creating advanced Tech Reborn machines and items, until you got to HV-Tier of Modern Industrialization. (and I'm glad this is no longer the case in SF3)

14.) Croparia as a whole - its only purpose is to get the Dragon Egg Seed, which is uncraftable, since the Ender Dragon only drops the Dragon Egg once - it was dropping multiple in version v 0.5.6 which didn't have the Progressive Bosses mod.
The various tiers of Croparias may seem expensive at first, but if you read the mod's guidebook or you see Croparia's CurseForge page, written in pure Engrish, you can see that you can just use the Elematilius Infusor to get the next tier of Croparia way cheaper.

15.) 🔧 "The key is behind the lock" - which is the Spectrum Stratine Gem's recipe. To unlock it, you need to, well, obtain a Stratine Gem in the first place, which is a rare drop from an unobtainable Stratine Ore. This make quite a bit of side content unobtainable.
(You can obtain the ore using the Ore Spatial Harvester, which you are not going to get until endgame) 

16.) Progression cheese:
You can find Uncrafting Tables in "large hill" Twilight Forest loot chests. Since the normal recipe requires an Ultimate Table, you can just uncraft the Uncrafting Table to get it for free. (Why uncrafting is enabled in the first place??? There is a reason almost every pack disables this functionality)

**Comment by MSandro** [comment]
Thanks for your help. Once I have some more time on the project again, I'll go through your list and fix it.

**Comment by ivanpu** [comment]
Tech Reborn ingots (from Alloy Forge): they're 1x1 craftable into MI ingots (isn't shown in REI, found by trial and error), and as a workaround can be processed in any "autocrafter" (I personally pass them thru Botania's Crafty Crate before piping into my storage system).

**Comment by ivanpu** [comment]
Also... is mixing (in MI mixer) any dye, with no other ingredient, into shulker box of the same color considered intended?

**Comment by KonSola5** [comment]
Definitely wrong.
![picture](https://user-images.githubusercontent.com/125081901/218516967-ff4d9a35-cba4-4911-ba18-a3164a4b3e8b.png)
Edit: Neither unintended, nor intended recipes work.
![picture2](https://user-images.githubusercontent.com/125081901/218517720-ebad9b6a-8a47-47cc-8926-f09a4837faaf.png)
![picture3](https://user-images.githubusercontent.com/125081901/218517796-be357f72-bf33-4cef-aa06-9ceb4e302b34.png)

**Comment by MSandro** [comment]
It requires the electric variant of the mixer. The fist recipe is just a kind of REI bug and doesn't really exists.

**Comment by KonSola5** [comment]
BTW, here's why the recipes for forging are incorrect: The default recipes have a priority depending on mods loaded. In the recipes, Tech Reborn has the highest priority, then IndRev, and MI has the lowest priority (why???)
![priority](https://user-images.githubusercontent.com/125081901/219356742-a507fd9f-7e39-4729-846d-59d30a0f075a.png)
After altering the priority, the recipes even show in REI!

**Comment by ivanpu** [comment]
Just Finished reading MI guidebook... and discovered replicator.
Not sure yet about requirements to automate UU-matter (I've just unlocked aluminium, and this is my first time with MI), but it can be piped into the machine, and the replicated items can be piped out.
Apparently it doesn't solve the dragon egg issue (the egg have a blacklist tag assigned), but it does "fix" these points:

> 5.) Record Fragments

Are blacklisted, but disc 13 is not.

> 6.) Ultimate Fish and Chips and Glitch Ingots.

Ultimate Fish and Chips - all ingredients are replicatable.
Glitch Ingot is replicatable (as the fragments to make it).

> 11.) Block of Fiery Metal

Replicatable, but probably no longer needed at this stage.

> 16.) Progression cheese: ... Uncrafting Tables

Replicatable... which _could_ make the issue worse, but at this stage you probably have industrial quantities of Neutronium anyways.


So, from my analysis you only will need 32 Infinity Catalysts to make your 1st replicator, and then only get one of the items above to start "automating" them.

Calculations:
32x ∞ :cat2: → 32x glitch ingot → 256x Physically-Condensed Matrix Fragment → 86 System Glitches (you'll be left with 2 spare fragments).
And also 5 samples of every fish plus 5 discs 13 (1 extra to start replicating) - but may still take long due to RNG.

In summary looks much more doable :)

Edit:
Block of terrasteel is blacklisted, but the ingonts not - looks like a loophole.

**Comment by KonSola5** [comment]
Nice analysis @ivanpu! Totaly missed the fact that the Replicator does not need any Infinity Ingots!
However, Tiny Piles of Neutrons require 100 mB of Helium Plasma each.
I myself got to the nuclear reactor and Plutonium production, and then the whole main MI pipe network stopped transferring items. Now need to rebuild the whole system again, but utilizing AE2 to split parts apart...

Will see, how viable is Helium Plasma Production, because it will need a good balance between using that for power and for other purposes... (In fact, for the Neutronium needed for the Replicator, you need about 154 buckets of Helium Plasma)

The most expensive item that could be replicated would be Compressed material Blocks.
(Not Nukes, because on average luck **1 Singularity produces 1 bucket of UU Matter**, and 1 Singularity requires 64 Nukes). So to calculate: 1 Singularity -> 10 replicated items.

Terrasteel Ingots can be automated anyway using Entropinnyum Mana production (That's an amazingly powerful flower).

Music Discs can be automated by making a Skeleton to shoot a Creeper down (vanilla mechanic).

**Comment by ivanpu** [comment]
> Will see, how viable is Helium Plasma Production

Helium Plasma Cells are replicatable :D

**Comment by KonSola5** [comment]
That's not supposed to be the case, since most of the things that can contain other things should be in blacklist. (TLDR: it's a oversight)

**Comment by KonSola5** [comment]
You would also be able to produce UU Matter infinitely that way (100 mB -> 1000 mB)

**Comment by ivanpu** [comment]
Well... from all buckets/cells that I've tried so far, only UU Matter can't be replicated

**Comment by KonSola5** [comment]
Fusion Reactor fueled with Deuterium and Tritium produced by two Nuclear Reactors seems to be way simpler anyway, no need for 64 nukes which require nuclear waste to craft.

**Comment by ivanpu** [comment]
Found a solution for the Dragon Eggs:
TR Creative Storage Unit - it requires 4 Infinity Ingots and 4 Infinity Catalysts (for a total of 48 Infinity Catalysts) - it's a bit more grind to make (128 System Glitches), but it instantly gives you virtually infinite amount of any item, including the Dragon Eggs.

Edit: this one can also "duplicate" UU Matter Buckets (probably really just _any_ item).

**Comment by KonSola5** [comment]
Have you considered, that for Cosmic Meatballs and Ultimate Fish and Chips you need Dragon Egg Fruit?

So, "the key is behind the lock"

But true: The TR Creative Storage Unit can duplicate anything.

The Creative ME Item Cell is even crazier, since you can literally drag any item from REI and get unlimited amount of that item, so that makes it a true "creative mode" instead of "journey mode".

**Comment by ivanpu** [comment]
> Have you considered, that for Cosmic Meatballs and Ultimate Fish and Chips you need Dragon Egg Fruit?

No, I've totally missed that...

**Comment by KonSola5** [comment]
There are way more "replicator_blacklist" oversights, like AE2 Cells beyond 64k, Coxinha Utilities Tanks and Hephaestus Tanks. The AE2 Cells are way bigger cheese since you can duplicate massive amounts of items/fluids from 100 mB of UU Matter.

Also, there are quite a bit of loopholes involving the Uncrafting Table, but since MSandro insists on keeping it enabled...

**Comment by ivanpu** [comment]
> BTW, here's why the recipes for forging are incorrect: The default recipes have a priority depending on mods loaded.  ... After altering the priority, the recipes even show in REI!

I've looked in the Alloy Forge files...
It looks like the same applies to other recipes (they work without being shown in REI), and specifically the "compat/alloys" category is worrying:
The worst is the steel, which can be made as "1x raw iron + 1x coal = 3x steel", which is much better than you'll ever get from MI EBF (1 iron → 1 steel) and way earlier than you even have steam.

Edit: indrev seems to have better rate than MI for steel, but its steel only 2 per iron and gated by extractinator.

Edit2: forgot that raw iron can produce 2 iron ingots (via the same AF) which makes:
* AF: 1 raw iron → 3 steel
* AF: 2 iron ingot → 4 steel
* MI: 2 iron ingot → 2 steel (but only 7 iron → 4 steel in the steam stage)
* IR: 2 iron ingot → 4 steel

So AF steel still feels out of order.

**Comment by KonSola5** [comment]
Did not know that Alloy Forgery could do steel at all.
And 2x raw ores already is a bit OP (even with new "Industrial Smeltery" mechanic, which is basically a modified IndRev ore processing chain, which is after getting MI Sulfuric Acid)

Would nerfing outputs to 1.33x (1.5x with End Stone Brick Forge) and yeeting steel altogether from Alloy Forge help?

Also about Stratine Gem: You can get this from one of the Ancient Ruins structures (there are 3 different ones), so if you really don't want to touch the Golden Drill, well you can.

**Comment by KonSola5** [comment]
Regarding fishing: Create Deployers can actually fish. But fishing in lava is still suspectible to RNG due to not being able to lavalog.

**Comment by ivanpu** [comment]
More about Alloy Forge:
Titanium and platinum ingots from raw ores feels like a progression cheese (requires Kanthal EBF otherwise).

And I think the above is the last of the problematic AF recipes:
All the AF alloys besides Steel are doable with the Create mixer, with only brass being better ratio than with Create.
And all the recipes for metals looks like doubling at max (when raw ore is even obtainable)... but it's your decision if it's too early for it.

**Comment by KonSola5** [comment]
Hephaestus can also do alloys.

However, if Platinum would be completely gated behind Kanthal EBF, then UPC (which requires Platinum) would be unviable, since once you get a Heat Exchanger, you are able to generate 7200 EU/t for basically free (lava).

While changing recipes, I made sure you can't melt Platinum Ore in Smeltery. (Ingots, dusts are still meltable)

**Comment by ivanpu** [comment]
Oh, forgot about smeltery... despite the fact that I've used it to skip the grind for the full prismarine alloy forge.

Apparently it have a similar issue with the recipes: **some do not show in the REI, but work.**
E.g. brass:
It shows what I can melt, and how I can alloy, but not that I can makes ingots from it.
And when I pour it into ingot cast - it produces Create brass ingot, which can be 1x1 crafted into TR ingot (again without hints in the REI).

Alloying in total shows 11 recipes, of which "obsidian from soup" appears to be broken both in REI (doesn't show the mushroom stew part) and in functioning (doesn't actually create obsidian).

_Offtopic:_
Overall I was a bit disappointed in this mod, compared to tconstruct in 1.12 (or at least with its integration in this modpack?):
While it does have some interesting automation opportunities, I couldn't find any way to get cobalt for most of T4 materials.
I haven't been to Eden Ring yet (and don't want to spoil it for myself before time) - but renforced tools are quite easy to get by automating ex-nihilo, and surpass Hephaestus T4 by far.
So even if cobalt is gated behind something that I didn't reach yet - it's too late for the tools progression.

**Comment by KonSola5** [comment]
Hephaestus has some internal bugs that the dev won't fix :(

Like if you want to search how to make a molten metal, TiC plugin will show every recipe.

Additionally, TiC itself is no longer an "OP tools mod" in 1.16+, you will eventually phase them out for Omni-Tool and Nanosaber.

If you have any ideas how to buff simple modifiers or materials to be better but not broken, let know.

And you are right - Croparia's tools OP (and MSandro still didn't do anything about them, seems like he's just not looking throughly what people push to him)

About materials in 0.6.0:
- Cobalt will be obtainable using a Gold Drill (MI), if it's too late, let know.
- There will be 3 new materials from Twilight Forest implemented.

But endless bugs and MSandro's lack of free time (aka busy IRL) still postpone the 0.6.0 update.

Be sure to check out progress in the "0.6.0" branch.

**Comment by KonSola5** [comment]
Correction on the post from 3 weeks ago: UPC does not require Platinum, it requires Iridium.

Anyways: Made yet another PR, in which I've corrected Alloy Forgery cheeses - most notably you can't obtain Steel, Aluminum and Platinum using Alloy Forge anymore.

**Comment by ivanpu** [comment]
> Be sure to check out progress in the "0.6.0" branch.

I've looked at some of the changes in 0.6.0... I think that the new hint (in Kibe quest chapter) that wither destroys blocks is irrelevant after removing Progressive Bosses.

Also, is there a way to convert it locally into a pack?
I see that the repo is not 1:1 relates to the modpack zip files.

And on topic, 0.5.7a - exctractinating rainbow asphalt gives much more than what's advertised:
![2023-04-10_02 03 59](https://user-images.githubusercontent.com/2607888/230800596-81c09abe-ef79-4be0-a5a3-b5079c318012.png)

**Comment by KonSola5** [comment]
You can install the prototype pack locally using packwiz (search it).
You will need to export the pack to a zip and install it using ex. PrismLauncher.

You can either clone and install what's on MSandro's 0.6.0 or clone my fork with extra changes.

**Comment by ivanpu** [comment]
> the whole main MI pipe network stopped transferring items

Happened to me too, but only partially, specifically with oak logs, which stopped rubber production, and as a result most of the automation...
Switching the "main storage bus" to Simple Storage pipes (and thus reducing MI pipes endpoints) fixed it for me, but I suspect only temporarily, until I extend the auto-crafting network (I leave the auto-crafted items in the machines, to not clutter the storage).
I don't understand yet how exactly motors affect the pipes besides moving more items in/out of the connected inventories (at least it didn't help with the logs), but I suspect that there might be some invisible limitations on the pipes throughput.
I also suspect that priorities might be a key for a pipes network built entirely from MI pipes (I only thought about it after changing to Simple Storage pipes) but so far I only used them to prevent the items from going into the wrong inventories.

**Comment by KonSola5** [comment]
> Oh, forgot about smeltery... despite the fact that I've used it to skip the grind for the full prismarine alloy forge.

The Grout is now gated behind Create's Mixer and Seared Bricks are made using the Alloy Forge.
